sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -1122,4 +1122,47 @@ class CachedTableSuite extends QueryTest with SQLTestUtils
       assert(!spark.catalog.isCached("t1"))
     }
   }
+
+  test("SPARK-30494 avoid duplicated cached RDD when replace an existing view") {
```

Comment:
This can be misleading because the cached RDD is not duplicated. Two cached RDD are independent.

Reply:
Ok. how about "Fix the leak of cached data when replace an existing view"